client/rest-high-level/src/main/java/org/elasticsearch/client/core/TermVectorsRequest.java

```diff
@@ -0,0 +1,230 @@
+        this.requestOffsets = requestOffsets;
+    }
+
+    /**
+     * Sets whether to request field statistics
+     * @param requestFieldStatistics
+     */
+    public void setFieldStatistics(boolean requestFieldStatistics) {
+        this.requestFieldStatistics = requestFieldStatistics;
+    }
+
+    /**
+     * Sets whether to request term statistics
+     * @param requestTermStatistics
+     */
+    public void setTermStatistics(boolean requestTermStatistics) {
+        this.requestTermStatistics = requestTermStatistics;
+    }
+
+    /**
+     * Sets different analyzers than the one at the fields
+     * @param perFieldAnalyzer
+     */
+    public void setPerFieldAnalyzer(Map<String, String> perFieldAnalyzer) {
+        this.perFieldAnalyzer = perFieldAnalyzer;
+    }
+
+    /**
+     * Sets an artifical document on what to request _termvectors
+     * @param docBuilder
+     */
+    public void setDoc(XContentBuilder docBuilder) {
+        this.docBuilder = docBuilder;
+    }
+
+    /**
+     * Sets conditions for terms filtering
+     * @param filterSettings
+     */
+    public void setFilterSettings(Map<String, Integer> filterSettings) {
+        this.filterSettings = filterSettings;
+    }
+
+    /**
+     * Sets a routing to route a request to a particular shard
+     * @param routing
+     */
+    public void setRouting(String routing) {
+        this.routing = routing;
+    }
+
+    public String getRouting() {
+        return routing;
+    }
+
+    /**
+     * Set a preference of which shard copies to execute the request
+     * @param preference
+     */
+    public void setPreference(String preference) {
+        this.preference = preference;
+    }
+
+    public String getPreference() {
+        return preference;
+    }
+
+    /**
+     * Sets if the request should be realtime or near-realtime
+     * @param realtime
+     */
+    public void setRealtime(boolean realtime) {
+        this.realtime = realtime;
+    }
+
+    public boolean getRealtime() {
+        return realtime;
+    }
+
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        // set values only when different from defaults
+        if (requestPositions == false) builder.field("positions", false);
+        if (requestPayloads == false) builder.field("payloads", false);
+        if (requestOffsets == false) builder.field("offsets", false);
+        if (requestFieldStatistics == false) builder.field("field_statistics", false);
+        if (requestTermStatistics) builder.field("term_statistics", true);
+        if (perFieldAnalyzer != null) builder.field("per_field_analyzer", perFieldAnalyzer);
+
+        if (docBuilder != null) {
+            BytesReference doc = BytesReference.bytes(docBuilder);
+            try (InputStream stream = doc.streamInput()) {
+                builder.rawField("doc", stream, docBuilder.contentType());
```

Comment:
I really don't like the name of this method! It makes me think we're copying the field raw without checking the content type but we *are* checking it.